scrapper.py

```diff
@@ -26,38 +34,120 @@ class Crawler:
+                if len(self.urls) < self.max_articles:
+                    self._extract_url(article_bs)
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
-
+        return self.seed_urls
+
+class HTMLParser:
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(article_url, article_id)
+
+    def _fill_article_with_text(self, article_bs):
+        text_bs = article_bs.find('div', class_='el-content uk-panel uk-margin-top')
+        self.article.text = text_bs.text
+
+    def _fill_article_with_meta_information(self, article_bs):
+        title_bs = article_bs.find('h2', class_='el-title uk-font-tertiary uk-margin-top uk-margin-remove-bottom').text
+        self.article.title = title_bs
+
+        date_bs = article_bs.find('div', class_='el-meta uk-text-meta uk-margin-top').text
+        months = {
+            'января':'01',
+            'февраля':'02',
+            'марта':'03',
+            'апреля':'04',
+            'мая':'05',
+            'июня':'06',
+            'июля':'07',
+            'августа':'08',
+            'сентября':'09',
+            'октября':'10',
+            'ноября':'11',
+            'декабря':'12'
+        }
+        for key, value in months.items():
+            date_bs = date_bs.replace(key, value)
+        date = datetime.strptime(date_bs, "%d %m %Y | %H:%M")
+        self.article.date = date
+
+
+        topic = article_bs.find_all('span', class_='uk-text-middle')
+        for hashtags in topic:
+            self.article.topics.append(hashtags.text)
+
+        self.article.author = 'NOT FOUND'
+
+    def parse(self):
+        response = requests.get(self.article_url)
+        article_bs = BeautifulSoup(response.text, 'html.parser')
+
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    try:
+        shutil.rmtree(base_path)
+    except FileNotFoundError:
+        pass
+    base_path.mkdir(parents=True)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path, 'r', encoding='utf-8') as file:
+        config = json.load(file)
+    seed_urls = config['seed_urls']
+    max_articles = config['total_articles_to_find_and_parse']
+    if not isinstance(max_articles, int) or max_articles <= 0:
+        raise IncorrectNumberOfArticlesError
+    if not isinstance(seed_urls, list) or not seed_urls:
+        raise IncorrectURLError
+    if max_articles > 200:
+        raise NumberOfArticlesOutOfRangeError
+    for urls in seed_urls:
+        if urls[0:8] != 'https://' and urls[0:7] != 'http://':
+            raise IncorrectURLError
+    return seed_urls, max_articles
```

Comment:
what if seed_urls is an empty list? would that still be considered correct, according to this function?

Reply:
Umm... isn't it checked by "if not seed_urls"?

Reply:
True :)